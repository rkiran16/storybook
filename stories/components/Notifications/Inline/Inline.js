import { html } from "lit-html";
import { Link } from "../../Link/Link";
import { IconButton } from "../../IconButton/IconButton";

const Icon = (type) => {
  switch (type) {
    case "warning":
      return html` <span class="material-icons "> warning_amber </span> `;
    case "success":
      return html`
        <span class="material-icons "> check_circle_outline </span>
      `;
    default:
      return html` <span class="material-icons "> error_outline </span> `;
  }
};

export const InlineNotification = ({
  title,
  label,
  color = "danger",
  icon = "default",
  href,
  linkTitle,
  href2,
  linkTitle2,
  dismiss = "true",
}) => {
  return html`
      <div
        class="alert alert-${color} border-${color} alert-dismissible d-flex justify-content-between align-items-center "
        role="alert"
      >

          <div class="d-flex">

          ${icon ? Icon(icon) : null}
          <div class="ms-2">
          <span>
            <strong class="text-${color}">${title}</strong>
            <span class="text-black text-break"> ${label}</span>
            <span class="ps-2">
              ${
                href
                  ? Link({
                      label: href && linkTitle ? linkTitle : "Link Action",
                      href: href,
                      variations: "secondary",
                    })
                  : null
              }
            </span>
            <span class="ps-2 ">
              ${
                href2
                  ? Link({
                      label: href2 && linkTitle2 ? linkTitle2 : "Link Action",
                      href: href2,
                      variations: "secondary",
                    })
                  : null
              }
            </span>
            </span>
            </div>
          </div>
 ${
   dismiss
     ? html` <button
         type="button"
         class="btn-close"
         data-bs-dismiss="alert"
         aria-label="Close"
       ></button>`
     : null
 }
         
        </div>
      </div>
  `;
};
//  ? html` <div>
//          ${IconButton({
//            textDirection: "vertical",
//            inverse: false,
//            disabled: false,
//            iconName: "close",
//            class: "p-0",
//          })}
//        </div>`
