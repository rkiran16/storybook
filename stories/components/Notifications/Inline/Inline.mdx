import { Canvas, Meta, Story } from "@storybook/addon-docs";

# Inline

Inline notifications are nondisruptive and confined to a specific area in the UI. Inline notifications display both task-generated and system-generated messages and persist until they are dismissed by the user or the notification is resolved. They are frequently used in conjunction with field-level messages for errors in forms or other input areas.

Message

- Message starts with “Error!”, “Warning!”, “Success!”, or “Info!” Depending on the state of the notification
- Keep the message under two lines
- Be descriptive and give users clear next steps

Sizing

- Width varies based on the context and page layout
- Height is based on the content length

Placement

- Place near their related items
- Do not cover other content
- In forms, place at the top or bottom

Dismissal

- Persist until the user dismisses them or takes action that resolves the notification
- Option for development to have notifications that can’t not be dismissed in a card

<Canvas>
  <Story id="components-notifications-inline--default"></Story>
</Canvas>

<Canvas>
  <Story id="components-notifications-inline--warning"></Story>
</Canvas>

<Canvas>
  <Story id="components-notifications-inline--success"></Story>
</Canvas>

<Canvas>
  <Story id="components-notifications-inline--info"></Story>
</Canvas>

<Canvas>
  <Story id="components-notifications-inline--with-action"></Story>
</Canvas>

<Canvas>
  <Story id="components-notifications-inline--with-no-dismissal"></Story>
</Canvas>

<Canvas>
  <Story id="components-notifications-inline--two-actions"></Story>
</Canvas>