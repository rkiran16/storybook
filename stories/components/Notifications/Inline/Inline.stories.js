import { InlineNotification } from "./Inline";
import docs from "./Inline.mdx";
import { html } from "lit-html";

export default {
  title: "Components/Notifications/Inline",
  parameters: {
    zeplinLink:
      "https://app.zeplin.io/project/62902a463701ffb0e532df2b/screen/6303d21f83d705123f9ece5b",
    docs: {
      page: docs,
    },
  },
  argTypes: {
    color: {
      name: "Color",
      control: { type: "select" },
      defaultValue: "danger",
      options: ["success", "info", "warning", "danger"],
    },
    icon: {
      name: "Icon",
      control: { type: "select" },
      defaultValue: "default",
      options: ["none", "default", "warning", "success"],
      mapping: {
        none: false,
        default: "same",
        warning: "warning",
        success: "success",
      },
    },
  },
};

const Template = (args) => html`${InlineNotification(args)}`;

export const Default = Template.bind({});
Default.args = {
  title: "Error!",
  label: "Lorem ipsum dolor sit amet, consectetur adipiscing elit",
};
export const Warning = Template.bind({});
Warning.args = {
  title: "Warning!",
  label: "Lorem ipsum dolor sit amet, consectetur adipiscing elit",
  color: "warning",
};

export const Success = Template.bind({});
Success.args = {
  title: "Success!",
  label: "Lorem ipsum dolor sit amet, consectetur adipiscing elit",
  color: "success",
};
export const Info = Template.bind({});
Info.args = {
  title: "Info!",
  label: "Lorem ipsum dolor sit amet, consectetur adipiscing elit",
  color: "info",
};

export const WithAction = Template.bind({});
WithAction.args = {
  title: "Error!",
  label: "Lorem ipsum dolor sit amet, consectetur adipiscing elit",
  href: "https://app.zeplin.io/project/62902a463701ffb0e532df2b/screen/6303d21f83d705123f9ece5b",
  linkTitle: "Link Action",
};

export const WithNoDismissal = Template.bind({});
WithNoDismissal.args = {
  title: "Error!",
  label: "Lorem ipsum dolor sit amet, consectetur adipiscing elit",
  dismiss: false,
};

export const TwoActions = Template.bind({});
TwoActions.args = {
  title: "Error!",
  label: "Lorem ipsum dolor sit amet, consectetur adipiscing elit",
  href: "https://app.zeplin.io/project/62902a463701ffb0e532df2b/screen/6303d21f83d705123f9ece5b",
  linkTitle: "Link Action",
  href2:
    "https://app.zeplin.io/project/62902a463701ffb0e532df2b/screen/6303d21f83d705123f9ece5b",
  linkTitle2: "Link Action",
};
