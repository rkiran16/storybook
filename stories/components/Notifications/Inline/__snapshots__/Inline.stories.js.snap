// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Storyshots Components/Notifications/Inline Default 1`] = `
Object {
  "_$litType$": 1,
  "strings": Array [
    "",
    "",
  ],
  "values": Array [
    Object {
      "_$litType$": 1,
      "strings": Array [
        "
      <div
        class=\\"alert alert-",
        " border-",
        " alert-dismissible d-flex justify-content-between align-items-center \\"
        role=\\"alert\\"
      >

          <div class=\\"d-flex\\">

          ",
        "
          <div class=\\"ms-2\\">
          <span>
            <strong class=\\"text-",
        "\\">",
        "</strong>
            <span class=\\"text-black text-break\\"> ",
        "</span>
            <span class=\\"ps-2\\">
              ",
        "
            </span>
            <span class=\\"ps-2 \\">
              ",
        "
            </span>
            </span>
            </div>
          </div>
 ",
        "
         
        </div>
      </div>
  ",
      ],
      "values": Array [
        "danger",
        "danger",
        Object {
          "_$litType$": 1,
          "strings": Array [
            " <span class=\\"material-icons \\"> error_outline </span> ",
          ],
          "values": Array [],
        },
        "danger",
        "Error!",
        "Lorem ipsum dolor sit amet, consectetur adipiscing elit",
        null,
        null,
        Object {
          "_$litType$": 1,
          "strings": Array [
            " <button
         type=\\"button\\"
         class=\\"btn-close\\"
         data-bs-dismiss=\\"alert\\"
         aria-label=\\"Close\\"
       ></button>",
          ],
          "values": Array [],
        },
      ],
    },
  ],
}
`;

exports[`Storyshots Components/Notifications/Inline Info 1`] = `
Object {
  "_$litType$": 1,
  "strings": Array [
    "",
    "",
  ],
  "values": Array [
    Object {
      "_$litType$": 1,
      "strings": Array [
        "
      <div
        class=\\"alert alert-",
        " border-",
        " alert-dismissible d-flex justify-content-between align-items-center \\"
        role=\\"alert\\"
      >

          <div class=\\"d-flex\\">

          ",
        "
          <div class=\\"ms-2\\">
          <span>
            <strong class=\\"text-",
        "\\">",
        "</strong>
            <span class=\\"text-black text-break\\"> ",
        "</span>
            <span class=\\"ps-2\\">
              ",
        "
            </span>
            <span class=\\"ps-2 \\">
              ",
        "
            </span>
            </span>
            </div>
          </div>
 ",
        "
         
        </div>
      </div>
  ",
      ],
      "values": Array [
        "info",
        "info",
        Object {
          "_$litType$": 1,
          "strings": Array [
            " <span class=\\"material-icons \\"> error_outline </span> ",
          ],
          "values": Array [],
        },
        "info",
        "Info!",
        "Lorem ipsum dolor sit amet, consectetur adipiscing elit",
        null,
        null,
        Object {
          "_$litType$": 1,
          "strings": Array [
            " <button
         type=\\"button\\"
         class=\\"btn-close\\"
         data-bs-dismiss=\\"alert\\"
         aria-label=\\"Close\\"
       ></button>",
          ],
          "values": Array [],
        },
      ],
    },
  ],
}
`;

exports[`Storyshots Components/Notifications/Inline Success 1`] = `
Object {
  "_$litType$": 1,
  "strings": Array [
    "",
    "",
  ],
  "values": Array [
    Object {
      "_$litType$": 1,
      "strings": Array [
        "
      <div
        class=\\"alert alert-",
        " border-",
        " alert-dismissible d-flex justify-content-between align-items-center \\"
        role=\\"alert\\"
      >

          <div class=\\"d-flex\\">

          ",
        "
          <div class=\\"ms-2\\">
          <span>
            <strong class=\\"text-",
        "\\">",
        "</strong>
            <span class=\\"text-black text-break\\"> ",
        "</span>
            <span class=\\"ps-2\\">
              ",
        "
            </span>
            <span class=\\"ps-2 \\">
              ",
        "
            </span>
            </span>
            </div>
          </div>
 ",
        "
         
        </div>
      </div>
  ",
      ],
      "values": Array [
        "success",
        "success",
        Object {
          "_$litType$": 1,
          "strings": Array [
            " <span class=\\"material-icons \\"> error_outline </span> ",
          ],
          "values": Array [],
        },
        "success",
        "Success!",
        "Lorem ipsum dolor sit amet, consectetur adipiscing elit",
        null,
        null,
        Object {
          "_$litType$": 1,
          "strings": Array [
            " <button
         type=\\"button\\"
         class=\\"btn-close\\"
         data-bs-dismiss=\\"alert\\"
         aria-label=\\"Close\\"
       ></button>",
          ],
          "values": Array [],
        },
      ],
    },
  ],
}
`;

exports[`Storyshots Components/Notifications/Inline Two Actions 1`] = `
Object {
  "_$litType$": 1,
  "strings": Array [
    "",
    "",
  ],
  "values": Array [
    Object {
      "_$litType$": 1,
      "strings": Array [
        "
      <div
        class=\\"alert alert-",
        " border-",
        " alert-dismissible d-flex justify-content-between align-items-center \\"
        role=\\"alert\\"
      >

          <div class=\\"d-flex\\">

          ",
        "
          <div class=\\"ms-2\\">
          <span>
            <strong class=\\"text-",
        "\\">",
        "</strong>
            <span class=\\"text-black text-break\\"> ",
        "</span>
            <span class=\\"ps-2\\">
              ",
        "
            </span>
            <span class=\\"ps-2 \\">
              ",
        "
            </span>
            </span>
            </div>
          </div>
 ",
        "
         
        </div>
      </div>
  ",
      ],
      "values": Array [
        "danger",
        "danger",
        Object {
          "_$litType$": 1,
          "strings": Array [
            " <span class=\\"material-icons \\"> error_outline </span> ",
          ],
          "values": Array [],
        },
        "danger",
        "Error!",
        "Lorem ipsum dolor sit amet, consectetur adipiscing elit",
        Object {
          "_$litType$": 1,
          "strings": Array [
            "<a href=",
            " target=",
            ">",
            "</a>",
          ],
          "values": Array [
            "https://app.zeplin.io/project/62902a463701ffb0e532df2b/screen/6303d21f83d705123f9ece5b",
            "_self",
            "Link Action",
          ],
        },
        Object {
          "_$litType$": 1,
          "strings": Array [
            "<a href=",
            " target=",
            ">",
            "</a>",
          ],
          "values": Array [
            "https://app.zeplin.io/project/62902a463701ffb0e532df2b/screen/6303d21f83d705123f9ece5b",
            "_self",
            "Link Action",
          ],
        },
        Object {
          "_$litType$": 1,
          "strings": Array [
            " <button
         type=\\"button\\"
         class=\\"btn-close\\"
         data-bs-dismiss=\\"alert\\"
         aria-label=\\"Close\\"
       ></button>",
          ],
          "values": Array [],
        },
      ],
    },
  ],
}
`;

exports[`Storyshots Components/Notifications/Inline Warning 1`] = `
Object {
  "_$litType$": 1,
  "strings": Array [
    "",
    "",
  ],
  "values": Array [
    Object {
      "_$litType$": 1,
      "strings": Array [
        "
      <div
        class=\\"alert alert-",
        " border-",
        " alert-dismissible d-flex justify-content-between align-items-center \\"
        role=\\"alert\\"
      >

          <div class=\\"d-flex\\">

          ",
        "
          <div class=\\"ms-2\\">
          <span>
            <strong class=\\"text-",
        "\\">",
        "</strong>
            <span class=\\"text-black text-break\\"> ",
        "</span>
            <span class=\\"ps-2\\">
              ",
        "
            </span>
            <span class=\\"ps-2 \\">
              ",
        "
            </span>
            </span>
            </div>
          </div>
 ",
        "
         
        </div>
      </div>
  ",
      ],
      "values": Array [
        "warning",
        "warning",
        Object {
          "_$litType$": 1,
          "strings": Array [
            " <span class=\\"material-icons \\"> error_outline </span> ",
          ],
          "values": Array [],
        },
        "warning",
        "Warning!",
        "Lorem ipsum dolor sit amet, consectetur adipiscing elit",
        null,
        null,
        Object {
          "_$litType$": 1,
          "strings": Array [
            " <button
         type=\\"button\\"
         class=\\"btn-close\\"
         data-bs-dismiss=\\"alert\\"
         aria-label=\\"Close\\"
       ></button>",
          ],
          "values": Array [],
        },
      ],
    },
  ],
}
`;

exports[`Storyshots Components/Notifications/Inline With Action 1`] = `
Object {
  "_$litType$": 1,
  "strings": Array [
    "",
    "",
  ],
  "values": Array [
    Object {
      "_$litType$": 1,
      "strings": Array [
        "
      <div
        class=\\"alert alert-",
        " border-",
        " alert-dismissible d-flex justify-content-between align-items-center \\"
        role=\\"alert\\"
      >

          <div class=\\"d-flex\\">

          ",
        "
          <div class=\\"ms-2\\">
          <span>
            <strong class=\\"text-",
        "\\">",
        "</strong>
            <span class=\\"text-black text-break\\"> ",
        "</span>
            <span class=\\"ps-2\\">
              ",
        "
            </span>
            <span class=\\"ps-2 \\">
              ",
        "
            </span>
            </span>
            </div>
          </div>
 ",
        "
         
        </div>
      </div>
  ",
      ],
      "values": Array [
        "danger",
        "danger",
        Object {
          "_$litType$": 1,
          "strings": Array [
            " <span class=\\"material-icons \\"> error_outline </span> ",
          ],
          "values": Array [],
        },
        "danger",
        "Error!",
        "Lorem ipsum dolor sit amet, consectetur adipiscing elit",
        Object {
          "_$litType$": 1,
          "strings": Array [
            "<a href=",
            " target=",
            ">",
            "</a>",
          ],
          "values": Array [
            "https://app.zeplin.io/project/62902a463701ffb0e532df2b/screen/6303d21f83d705123f9ece5b",
            "_self",
            "Link Action",
          ],
        },
        null,
        Object {
          "_$litType$": 1,
          "strings": Array [
            " <button
         type=\\"button\\"
         class=\\"btn-close\\"
         data-bs-dismiss=\\"alert\\"
         aria-label=\\"Close\\"
       ></button>",
          ],
          "values": Array [],
        },
      ],
    },
  ],
}
`;

exports[`Storyshots Components/Notifications/Inline With No Dismissal 1`] = `
Object {
  "_$litType$": 1,
  "strings": Array [
    "",
    "",
  ],
  "values": Array [
    Object {
      "_$litType$": 1,
      "strings": Array [
        "
      <div
        class=\\"alert alert-",
        " border-",
        " alert-dismissible d-flex justify-content-between align-items-center \\"
        role=\\"alert\\"
      >

          <div class=\\"d-flex\\">

          ",
        "
          <div class=\\"ms-2\\">
          <span>
            <strong class=\\"text-",
        "\\">",
        "</strong>
            <span class=\\"text-black text-break\\"> ",
        "</span>
            <span class=\\"ps-2\\">
              ",
        "
            </span>
            <span class=\\"ps-2 \\">
              ",
        "
            </span>
            </span>
            </div>
          </div>
 ",
        "
         
        </div>
      </div>
  ",
      ],
      "values": Array [
        "danger",
        "danger",
        Object {
          "_$litType$": 1,
          "strings": Array [
            " <span class=\\"material-icons \\"> error_outline </span> ",
          ],
          "values": Array [],
        },
        "danger",
        "Error!",
        "Lorem ipsum dolor sit amet, consectetur adipiscing elit",
        null,
        null,
        null,
      ],
    },
  ],
}
`;
